rotkehlchen/api/rest.py

```diff
@@ -5533,28 +5533,34 @@ def get_historical_prices_per_asset(
             if (ts := Timestamp(from_timestamp + (i * interval))) not in exclude_timestamps
         ]
         if only_cache_period is not None:
-            # try special assets first: they use custom logic not stored in global DB
-            for ts in timestamps:
-                special_asset_price = PriceHistorian().get_price_for_special_asset(
-                    from_asset=asset,
-                    to_asset=main_currency,
-                    timestamp=ts,
-                    max_seconds_distance=only_cache_period,
-                )
-                if special_asset_price is not None:
-                    prices[ts] = str(special_asset_price)
+            if (  # check for manual current price first to avoid fluctuating USD conversion
+                    (manual_price_info := GlobalDBHandler.get_manual_current_price(asset)) is not None and  # noqa: E501
+                    manual_price_info[0] == main_currency
+            ):
+                prices = dict.fromkeys(timestamps, str(manual_price_info[1]))
```

Comment:
I believe this won't solve the issue. For example imagine and asset where I change the price every day and record it in the snapshot. Say a stock or some other kind of asset. Then this will break the historical of prices right?